### DISCUSSION TITLE
Simulated SLM and camera different pixel properties

### DISCUSSION THREAD

**Discussion by lufermar** [slmsuite/slmsuite#45] [Q&A] [answered]
Hello again,
I am trying to simulate some SLM-camera experiments to play with the code and in the computational example you define the same pixel numbers and pixel pitch. I tried to make them different but my far_field and near_field plots have the same pixel size, corresponding to the SLM (so it looks like my camera has the same pixel properties as the SLM). I know that the command `cameraslm` works for the hardware code, but I am missing something for the simulations. Could you give me a hint on how to proceed?

Best,
L

**Comment by ichristen** [comment, marked as answer]
Hi @lufermar ,

The padded SLM nearfield and SLM farfield will have the same pixel number, as these are related by a FFT. The camera field of view is a separate measure and will have some overlap with the SLM farfield, such as the below image from [the readthedocs examples](https://slmsuite.readthedocs.io/en/latest/_examples/experimental_holography.html#Pictorial-Holography). This is because practical alignment of the camera and SLM is, naturally, not a perfect correspondence. Experimental measurement of this mapping between the camera and SLM coordinates is done via a `FourierSLM` through the `fourier_calibrate()` function.

![Example](https://slmsuite.readthedocs.io/en/latest/_images/_examples_experimental_holography_72_0.png)

In the `main` branch, one cannot currently simulate the above transformations or calibrations. However, @cpanuski has been working on the `sim` branch which is able to simulate all of this stuff without hardware. You can take a look at [this example](https://github.com/QPG-MIT/slmsuite-examples/blob/main/examples/simulation.ipynb) which is not yet on readthedocs, but is the test notebook for the `sim` branch. This may be on `main` within a month or so (#15), but in the meantime, you can checkout the `sim` branch locally to try out these features.

Hope this helps!
